staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/controller.go

```diff
@@ -364,9 +365,13 @@ func (c *celAdmissionController) Validate(
 	}
 
 	if len(deniedDecisions) > 0 {
-		return &policyError{
-			deniedDecisions: deniedDecisions,
+		// TODO: refactor NewForbidden so the name extraction is reusable for different error types
+		err := admission.NewForbidden(a, errors.New(deniedDecisions[0].message)).(*k8serrors.StatusError)
```

Comment:
Doesn't this drop the reason field?

Reply:
Also, I believe all the tests (esp. integration) need to be updated to expect whatever message format is returned here.

Reply:
> Doesn't this drop the reason field?

that's stomped in in the following lines

Reply:
The reason be reset to correct one following. Since we will return the same message, if integration test checks if certain message are contained, it should be fine

Reply:
Update: I handled the reason default to invalid here so hopefully it works as expected.